src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java

```diff
@@ -751,24 +981,143 @@ public ApprovedWhitelist() {
+    @VisibleForTesting
+    @Restricted(NoExternalUse.class)
+    synchronized void denyScript(String hash) {
         Jenkins.getInstance().checkPermission(Jenkins.RUN_SCRIPTS);
         approvedScriptHashes.remove(hash);
         removePendingScript(hash);
         save();
     }
 
-    private synchronized void removePendingScript(String hash) {
+    @RequirePOST
+    @Restricted(NoExternalUse.class)
+    public void doScriptContent(StaplerRequest req, StaplerResponse rsp, @QueryParameter(required = true) String hash) throws Exception {
+        Jenkins.getInstance().checkPermission(Jenkins.RUN_SCRIPTS);
+
+        synchronized (this) {
+            FullScriptMetadata metadata = this.metadataStorage.getExisting(hash);
+            if (metadata == null) {
+                // this can occur naturally only if you have concurrent views of the scriptApproval page
+                LOG.log(Level.FINER, "Requesting a non-existing metadata {0}", hash);
+                throw HttpResponses.notFound();
+            }
+            Language language = metadata.getLanguage();
+            if (language != null && metadata.getScriptLength() > 0) {
+                String script = this.metadataStorage.readScript(hash);
+                if (script != null) {
+                    req.setAttribute("script", script);
+                    req.setAttribute("languageCodeMirrorMode", language.getCodeMirrorMode());
+                    req.getView(this, "tabs/_scriptContent.jelly").forward(req, rsp);
+                    return;
+                } else {
+                    // expected for legacy approved scripts (until first usage)
+                    LOG.log(Level.FINER, "Requesting a non-existing script {0}", hash);
+                }
+            }
+            throw HttpResponses.notFound();
+        }
+    }
+
+    @Restricted(NoExternalUse.class)
+    public static final class AllSelectedHashesModel {
+        public String[] hashes;
+    }
+
+    @RequirePOST
+    @Restricted(NoExternalUse.class)
+    public void doApprovePendingScripts(@JsonBody AllSelectedHashesModel content) {
+        Jenkins.getInstance().checkPermission(Jenkins.RUN_SCRIPTS);
+        LOG.log(Level.FINE, "Approval granted for selected {0}", content.hashes);
+
+        synchronized (this) {
+            String userLogin = this.getCurrentUserLogin();
+
+            for (String hash : content.hashes) {
+                approvedScriptHashes.add(hash);
+                PendingScript pendingScript = removePendingScript(hash);
+
+                if (METADATA_GATHERING) {
+                    String script = pendingScript == null ? null : pendingScript.script;
+                    this.metadataStorage.withMetadata(hash, script, metadata ->
+                            metadata.notifyApproval(pendingScript, userLogin)
+                    );
+                }
+            }
+
+            save();
+        }
+        SecurityContext orig = ACL.impersonate(ACL.SYSTEM);
```

Comment:
This is deprecated! Use `ACL.as`, and fix the existing two usages in this class while you are at it.